libs/tokens/src/hooks/tokens/useTokenBySymbolOrAddress.ts

```diff
@@ -2,30 +2,61 @@ import { useAtomValue } from 'jotai'
 import { useMemo } from 'react'
 
 import { TokenWithLogo } from '@cowprotocol/common-const'
+import { getAddress } from '@ethersproject/address'
 
 import { useTokensByAddressMap } from './useTokensByAddressMap'
 
+import { environmentAtom } from '../../state/environmentAtom'
 import { tokensBySymbolAtom } from '../../state/tokens/allTokensAtom'
+import { userAddedTokensAtom } from '../../state/tokens/userAddedTokensAtom'
 
 export function useTokenBySymbolOrAddress(symbolOrAddress?: string | null): TokenWithLogo | null {
   const tokensByAddress = useTokensByAddressMap()
   const tokensBySymbol = useAtomValue(tokensBySymbolAtom).tokens
+  // Get user added tokens state
+  const userAddedTokens = useAtomValue(userAddedTokensAtom)
+  const { chainId } = useAtomValue(environmentAtom)
 
   return useMemo(() => {
-    if (!symbolOrAddress) {
+    if (!symbolOrAddress || !chainId) {
       return null
     }
 
     const symbolOrAddressLowerCase = symbolOrAddress.toLowerCase()
+    const userAddedTokensForChain = userAddedTokens[chainId] || {}
+
+    // Check if it's an address in the userAddedTokens map first
+    try {
+      const checksummedAddress = getAddress(symbolOrAddress)
+      const tokenInfo = userAddedTokensForChain[checksummedAddress.toLowerCase()]
```

Comment:
Why checksum the address if you lowercase it?
I guess you are using it only for checking whether it's a valid address?

If that's what you want, you can achieve the same without the extra step and the exception handling with [isAddress](https://docs.ethers.org/v5/api/utils/address/#utils-isAddress)

Reply:
Thank you, addressed now.